### PR TITLE
Improve actions workflow

### PR DESCRIPTION
**Pull request by iKlsR**
- Use master where possible to keep up to date with actions
- Verbose step names
- Use a cache for deps
- Yarn comes preinstalled in os images

Test this I guess

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
 # For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
 
-name: Build
+name: Update and build application
```
Comment (njbrown) [inline .github/workflows/build.yml:4]: Shouldn't the name be more of a title? Like "Update and Build Application"

Comment (iKlsR) [reply .github/workflows/build.yml:4]: It's a sentence, not a blog title

Comment (iKlsR) [reply .github/workflows/build.yml:4]: It's a sentence, not a blog title imo but your personal flavor may differ

**Comment by njbrown** [comment]
Looks great 👍

**Pull request merged**